CardsCashCasino/Manager/TexasHoldEmManager.cs

```diff
@@ -0,0 +1,172 @@
+ *  Known Faults: None encountered
+ */
+
+using System;
+using System.Collections.Generic;
+using CardsCashCasino.Data;
+
+namespace CardsCashCasino.Manager
+{
+    /// <summary>
+    /// The possible actions that a player can take in Texas Hold Em.
+    /// </summary>
+    public enum PokerAction
+    {
+        FOLD,
+        CHECK,
+        CALL,
+        RAISE,
+        ALL_IN
+    }
+
+    /// <summary>
+    /// The possible roles that a player can have in Texas Hold Em.
+    /// The Small Blind is always left of the Dealer.
+    /// </summary>
+    public enum PlayerRole
+    {
+        DEALER,
+        SMALL_BLIND,
+        BIG_BLIND,
+    }
+    
+    public class TexasHoldEmManager
+    {
+        /// <summary>
+        /// The cardManager for the game.
+        /// </summary>
+        private CardManager _cardManager;
+        
+        /// <summary>
+        /// The user's hand of cards.
+        /// </summary>
+        private UserHand _userHand;
+        
+        /// <summary>
+        /// Flag that indicates if the user is still active in the game.
+        /// </summary>
+        private bool _gameOver;
+        
+        /// <summary>
+        /// The chip manager for the game.
+        /// </summary>
+        private ChipManager _chipManager;
+        
+        /// <summary>
+        /// The main pot for the game.
+        /// </summary>
+        private int _mainPot;
+        
+        /// <summary>
+        /// The side pot for the game.
+        /// </summary>
+        private int _sidePot;
+        
+        /// <summary>
+        /// The current bet for the game.
+        /// This is the amount that all players must match to stay in the game.
+        /// </summary>
+        private int _currentBet;
+        
+        /// <summary>
+        /// The Small Blind for the game.
+        /// </summary>
+        private int _smallBlind;
+        
+        /// <summary>
+        /// The Big Blind for the game.
+        /// </summary>
+        private int _bigBlind;
+        
+        /// <summary>
+        /// The Ante for the game.
+        /// This is the amount that all players must pay to play the game.
+        /// </summary>
+        private int _ante;
+        
+        /// <summary>
+        /// A dictionary that tracks which player has what role.
+        /// </summary>
+        private Dictionary<PlayerRole, int> _players { get; set; }
+        
+        private List<CardHand> _playerHands { get; set; }
+
+        public TexasHoldEmManager()
+        {
+            _cardManager = new CardManager();
+            _userHand = new UserHand();
+            _gameOver = false;
+            _chipManager = new ChipManager();
+            _mainPot = 0;
+            _sidePot = 0;
+            _currentBet = 0;
+            _smallBlind = 0;
+            _bigBlind = 0;
+            _ante = 0;
+
+            // Initialize the player hands. The first player is the user.
+            for (int i = 0; i < Constants.AI_PLAYER_COUNT; i++)
```

Comment:
I see the intuition behind including the UserHand in this list, but I think it will make some things way harder especially as we have to make the list use the abstract class instead of the inherited class.

I am not sure the ideal solution to this, but I think we will benefit from handling the Draw / Update for UserHand at a separate step than the PokerHand classes, because they will be drawn differently. The two, rather flawed, ideas I had were to create a "nextHand" pointer in either OR keep track of when the user plays in the order. I am sure there is a much better way to solve that problem, but it is probably worth some discussion in the Discord.

Reply:
I figured that since both UserHand and PokerHand are derived from the abstract class, so we should be able to be acted on in the same way that the PokerHand class would, right? Can you explain more about what the complication would be? 

And maybe I'm not thinking about this correctly, but would we have to implement the Draw /Update methods in different places between two hands as long as the logic is different? I'm thinking of something that was similar to how we drew the ships according to whether or not they were sunk or whose ships they belonged to, except with if all players are all in, what part of the round it is, etc.